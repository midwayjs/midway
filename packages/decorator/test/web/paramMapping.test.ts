import { expect } from 'chai';
import {
  ALL,
  Body,
  File,
  Files,
  getPropertyDataFromClass,
  Headers,
  Param,
  Query,
  Session,
  WEB_ROUTER_PARAM_KEY,
} from '../../src';

class Test {
  async doget(
    @Param('aa') aa: any,
    @Query('bb') query: any,
    @Body('body') body: any,
    @Headers('tt') tt: any,
    @File({ requireFile: true }) f: any,
    @Files() files: any,
    @Session(ALL) bb: any
  ) {}
}

describe('/test/web/paramMapping.test.ts', () => {
  it('paramMapping decorator should be ok', () => {
    const meta = getPropertyDataFromClass(WEB_ROUTER_PARAM_KEY, Test, 'doget');
    expect(meta.length).eq(7);
  });
});
