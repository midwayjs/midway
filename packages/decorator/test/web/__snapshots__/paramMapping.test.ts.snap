// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`/test/web/paramMapping.test.ts paramMapping decorator should be ok 1`] = `
{
  "doget": [
    {
      "key": "web:router_param",
      "metadata": {
        "propertyData": "common:all_value_key",
        "type": "session",
      },
      "options": {
        "impl": true,
        "pipes": undefined,
      },
      "parameterIndex": 6,
      "propertyName": "doget",
    },
    {
      "key": "web:router_param",
      "metadata": {
        "propertyData": undefined,
        "type": "files_stream",
      },
      "options": {
        "impl": true,
        "pipes": undefined,
      },
      "parameterIndex": 5,
      "propertyName": "doget",
    },
    {
      "key": "web:router_param",
      "metadata": {
        "propertyData": {
          "requireFile": true,
        },
        "type": "file_stream",
      },
      "options": {
        "impl": true,
        "pipes": undefined,
      },
      "parameterIndex": 4,
      "propertyName": "doget",
    },
    {
      "key": "web:router_param",
      "metadata": {
        "propertyData": "tt",
        "type": "headers",
      },
      "options": {
        "impl": true,
        "pipes": undefined,
      },
      "parameterIndex": 3,
      "propertyName": "doget",
    },
    {
      "key": "web:router_param",
      "metadata": {
        "propertyData": "body",
        "type": "body",
      },
      "options": {
        "impl": true,
        "pipes": undefined,
      },
      "parameterIndex": 2,
      "propertyName": "doget",
    },
    {
      "key": "web:router_param",
      "metadata": {
        "propertyData": "bb",
        "type": "query",
      },
      "options": {
        "impl": true,
        "pipes": undefined,
      },
      "parameterIndex": 1,
      "propertyName": "doget",
    },
    {
      "key": "web:router_param",
      "metadata": {
        "propertyData": "aa",
        "type": "param",
      },
      "options": {
        "impl": true,
        "pipes": undefined,
      },
      "parameterIndex": 0,
      "propertyName": "doget",
    },
  ],
}
`;
